datafusion/common/src/scalar/mod.rs

```diff
@@ -4539,7 +4539,8 @@ mod tests {
         // The alignment requirements differ across architectures and
         // thus the size of the enum appears to as well
 
-        assert_eq!(std::mem::size_of::<ScalarValue>(), 48);
+        // The value can be changed depending on rust version
+        assert_eq!(std::mem::size_of::<ScalarValue>(), 64);
```

Comment:
This is an interesting change -- basically it means that all `ScalarValue` based code will potentially consume significantly more memory (as now each ScalarValue takes 64 bytes rather than 48). Interestingly, this is the same behavior as aarch64 in previous rust versions (you can see this test is cfg'd off with

```rust
    #[cfg(not(target_arch = "aarch64"))]
```

I'll make a follow on PR to remove this cfg
```

Reply:
https://github.com/apache/arrow-datafusion/pull/9728